comfy_panel/special_games.lua

```diff
@@ -0,0 +1,199 @@
+local Event = require 'utils.event'
+local Color = require 'utils.color_presets'
+local valid_special_games = {
+	turtle = {
+		name = "Turtle",
+		config = {
+			[1] = {name = "label1", type = "label", caption = "moat width"},
+			[2] = {name = 'moat_width', type = "textfield", text = "3", numeric = true, width = 40},
+			[3] = {name = "label2", type = "label", caption = "entrance width"},
+			[4] = {name = 'entrance_width', type = "textfield", text = "20", numeric = true, width = 40},
+			[5] = {name = "label3", type = "label", caption = "size x"},
+			[6] = {name = 'size_x', type = "textfield", text = "200", numeric = true, width = 40},
+			[7] = {name = "label4", type = "label", caption = "size y"},
+			[8] = {name = 'size_y', type = "textfield", text = "200", numeric = true, width = 40}
+
+		},
+		button = {name = "turtle_apply", type = "button", caption = "Apply"}
+	},
+	infinity_chest = {
+		name = "Infinity chest",
+		config = {
+			[1] = {name = "separate_chests", type = "checkbox", caption = "Separate chest for each item", state = false},
+			[2] = {name = "eq1", type = "choose-elem-button", elem_type = "item"},
+			[3] = {name = "eq2", type = "choose-elem-button", elem_type = "item"},
+			[4] = {name = "eq3", type = "choose-elem-button", elem_type = "item"},
+			[5] = {name = "eq4", type = "choose-elem-button", elem_type = "item"},
+			[6] = {name = "eq5", type = "choose-elem-button", elem_type = "item"}
+		},
+		button = {name = "infinity_chest_apply", type = "button", caption = "Apply"}
+	}
+}
+
+local function generate_turtle(moat_width, entrance_width, size_x, size_y)
+	game.print("Special game turtle is being generated!", Color.warning)
+	local surface = game.surfaces[global.bb_surface_name]
+	local water_positions = {}
+	local concrete_positions = {}
+	local k = 1
+	for i = 1, size_y + moat_width do
+		for a = 0, moat_width do
+			table.insert(water_positions, {name = "water", position = {x = (size_x / 2) + a, y = i}}) -- north
+			table.insert(water_positions, {name = "water", position = {x = -(size_x / 2) - a, y = i}})
+			table.insert(water_positions, {name = "water", position = {x = (size_x / 2) + a, y = -i}}) -- south
+			table.insert(water_positions, {name = "water", position = {x = -(size_x / 2) - a, y = -i}})
+		end
+	end
+	for i = 1, size_x do
+		for a = 0, moat_width do
+			table.insert(water_positions, {name = "water", position = {x = i - (size_x / 2), y = size_y + a}}) -- north 
+			table.insert(water_positions, {name = "water", position = {x = i - (size_x / 2), y = -size_y - a}}) -- south
```

Comment:
If you use deepwater in place of water, your moat color would nicely blend into the river.

```suggestion
			table.insert(water_positions, {name = "deepwater", position = {x = (size_x / 2) + a, y = i}}) -- north
			table.insert(water_positions, {name = "deepwater", position = {x = -(size_x / 2) - a, y = i}})
			table.insert(water_positions, {name = "deepwater", position = {x = (size_x / 2) + a, y = -i}}) -- south
			table.insert(water_positions, {name = "deepwater", position = {x = -(size_x / 2) - a, y = -i}})
		end
	end
	for i = 1, size_x do
		for a = 0, moat_width do
			table.insert(water_positions, {name = "deepwater", position = {x = i - (size_x / 2), y = size_y + a}}) -- north 
			table.insert(water_positions, {name = "deepwater", position = {x = i - (size_x / 2), y = -size_y - a}}) -- south
```